pkg/bootstrap/node.go

```diff
@@ -687,3 +692,86 @@ func ensureSelinux() {
 		log.Fatalln("ERROR: SELinux enforcing mode is enabled. You will need to disable it with 'sudo setenforce 0' for kube-spawn to work properly.")
 	}
 }
+
+func checkCoreosSemver(coreosVer string) error {
+	v, err := semver.NewVersion(coreosVer)
+	if err != nil {
+		return err
+	}
+
+	c, err := semver.NewConstraint(">=" + coreosStableVersion)
+	if err != nil {
+		log.Printf("cannot get constraint for >= %s: %v", coreosStableVersion, err)
+		return err
+	}
+
+	if c.Check(v) {
+		return nil
+	} else {
+		return fmt.Errorf("ERROR: Container Linux version %s is too low in your local image.", coreosVer)
+	}
+}
+
+func checkCoreosVersion() error {
+	args := []string{
+		"image-status",
+		"coreos",
+	}
+
+	cmd := exec.Command("machinectl", args...)
+	cmd.Stderr = os.Stderr
+	cmd.Stdin = os.Stdin
+
+	b, err := cmd.Output()
+	if err != nil {
+		return err
+	}
+
+	checkCoreosVersionField := func(values []string) error {
+		for _, v := range values {
+			if err := checkCoreosSemver(strings.TrimSpace(v)); err != nil {
+				if err == semver.ErrInvalidSemVer {
+					// just meaning it's not a version field, so continue to the next field
+					continue
+				} else {
+					return err
+				}
+			} else {
+				return nil
+			}
+		}
+		return fmt.Errorf("cannot find a version field")
+	}
+
+	s := bufio.NewScanner(strings.NewReader(string(b)))
+	for s.Scan() {
+		// an example line from machinectl image-status:
+		//  OS: Container Linux by CoreOS 1478.0.0 (Ladybug)
+
+		line := strings.Split(s.Text(), ":")
+		if len(line) <= 1 {
+			continue
+		}
+
+		keyStr := strings.TrimSpace(line[0])
+		valueStr := strings.TrimSpace(line[1])
+		if keyStr != "OS" {
+			continue
+		}
+
+		// now the line has the key "OS", so get the version field in the values
+		values := strings.Fields(valueStr)
+		if err := checkCoreosVersionField(values); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func ensureCoreosVersion() {
+	if err := checkCoreosVersion(); err != nil {
+		log.Println(err)
+		log.Fatalf("You will need to remove the image by 'sudo machinectl remove coreos' and download the version %s at least by 'sudo machinectl pull-raw --verify=no https://alpha.release.core-os.net/amd64-usr/current/coreos_developer_container.bin.bz2 coreos'.", coreosStableVersion)
```

Comment:
you could recommend the user to remove the old image and let `kube-spawn` downloads a new one automatically.